### PR TITLE
vimPlugins: vim-go: make gocode, gocode-mod and keyify available to it

### PR DESCRIPTION
**Pull request by kalbasit**
###### Motivation for this change

This is a continuation of #49669, it seems that I have missed the gocode-gomod binary and I forgot to add gocode and go-tools to the vim-go path.

With regards to the `sed` change, initially, I patched [go#path#CheckBinPath()](https://github.com/fatih/vim-go/blob/5fffae94cbb45f3f5452fe423011729499f5389b/autoload/go/path.vim#L154) which is called anytime a binary is called. However, there's also [this call](https://github.com/fatih/vim-go/blob/5fffae94cbb45f3f5452fe423011729499f5389b/plugin/go.vim#L93) left unpached that can cause confusion if the user runs `:GoInstallBinaries` as it won't see the nix-store paths. In this PR, I decided to patch [go#config#BinPath()](https://github.com/fatih/vim-go/blob/5fffae94cbb45f3f5452fe423011729499f5389b/autoload/go/config.vim#L351-L353) which is called by any call out to a binary. As an added bonus, this patch patches the default go_bin_path, so a user can simply `let g:go_bin_path="..."` from the vimrc to override it.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @Mic92

### CODE REVIEW

File: pkgs/misc/vim-plugins/overrides.nix [hunk 1]
```diff
@@ -261,6 +261,9 @@ with generated;
       asmfmt
       delve
       errcheck
+      go-tools
+      gocode
+      gocode-gomod
```
Comment (Mic92) [inline pkgs/misc/vim-plugins/overrides.nix]: What does gocode-gomode?

Comment (kalbasit) [reply pkgs/misc/vim-plugins/overrides.nix]: I was taken by this as well. gocode-gomod is a fork of gocode with support for Go modules. Unfortunately, neither project can replace the other!

See gocode README https://github.com/mdempsky/gocode#githubcommdempskygocode

Comment (Mic92) [reply pkgs/misc/vim-plugins/overrides.nix]: Is vim-go smart enough to switch between the too?

Comment (kalbasit) [reply pkgs/misc/vim-plugins/overrides.nix]: yes, it is. it installs it [here](
https://github.com/fatih/vim-go/blob/8513febab5c698553acc7ac33f4185fb2da664d2/plugin/go.vim#L50) and uses it [](
https://github.com/fatih/vim-go/blob/ab0eb131ae5f3e2b7c069291488d6636cc717843/autoload/go/complete.vim#L14).

**Pull request merged**